replay/models/knn.py

```diff
@@ -20,16 +20,19 @@ def _get_ann_infer_params(self) -> Dict[str, Any]:
             "index_type": "sparse",
         }
 
-    def _get_vectors_to_infer_ann(self, log: DataFrame, users: DataFrame) -> DataFrame:
-        user_to_max_items = (
-            log.groupBy('user_idx')
-            .agg(sf.count('item_idx').alias('num_items'))
-        )
-        users = users.join(user_to_max_items, on="user_idx")
+    def _get_vectors_to_infer_ann(
```

Comment:
It seems "_get_vectors_to_infer_ann", "_get_ann_build_params", "_get_vectors_to_build_ann" are full duplicates of what you have in slim. Having both knn and slim are descendents of the same base class 'NeughborRec', pls, consider moving these functions to this base class.